typescript/ccip-server/src/services/LightClientService.ts

```diff
@@ -0,0 +1,90 @@
+// @ts-nocheck
+import axios from 'axios';
+import { ethers, utils } from 'ethers';
+
+import { TelepathyCcipReadIsmAbi } from '../abis/TelepathyCcipReadIsmAbi';
+
+import { Requestor } from './common/Requestor';
+
+enum ProofStatus {
+  running = 'running',
+  success = 'success',
+}
+
+// Service that interacts with the LightClient/ISM
+class LightClientService extends Requestor {
+  // Stores the current ProofId that is being generated. Clears once proof is ready.
+  pendingProofId: string;
+
+  constructor(
+    private readonly lightClientContract: ethers.Contract,
```

Comment:
might be nice to use typechain contracts here like we do in the sdk

Reply:
Will fix in another PR